src/modules/claims/claims.service.ts

```diff
@@ -843,6 +877,198 @@ export class ClaimsService {
     return claimType.split('.roles.')[1];
   }
 
+  /**
+   * Revoke On-Chain issued claim by `claimId` or given `namespace` and `subject`. Required `claimId` or `claim` parameters.
+   *
+   * ```typescript
+   * claimsService.revokeClaim({
+   *     claim: {
+   *         namespace: 'root.roles.energyweb.iam.ewc',
+   *         subject: 'did:ethr:volta:0x00...0',
+   *     },
+   *     registrationTypes = [RegistrationTypes.OnChain, RegistrationTypes.OffChain],
+   * });
+   * ```
+   * or
+   * ```typescript
+   * claimsService.revokeClaim({
+   *     claimId: claim.id,
+   *     registrationTypes = [RegistrationTypes.OnChain, RegistrationTypes.OffChain],
+   * });
+   * ```
+   *
+   * @param {RevokeClaimOptions} options object containing options
+   * @return true if claim was revoked
+   */
+  async revokeClaim(options: RevokeClaimOptions): Promise<boolean> {
+    const revoker = this._signerService.did;
+    const { namespace, subject } = await this.getRevocationClaimDetails(
+      options
+    );
+
+    return await this._claimRevocation.revokeClaim(namespace, subject, revoker);
+  }
+
+  /**
+   * Revoke On-Chain issued claims of the given namespace for multiple subjects. Namespace must be the same for all subjects.
+   *
+   * ```typescript
+   * claimsService.revokeMultipleClaim({
+   *     claims: [{
+   *         namespace: 'root.roles.energyweb.iam.ewc',
+   *         subject: 'did:ethr:volta:0x00...0',
+   *         registrationTypes = [RegistrationTypes.OnChain, RegistrationTypes.OffChain],
+   *     },
+   *     {
+   *         namespace: 'root.roles.energyweb.iam.ewc',
+   *         subject: 'did:ethr:volta:0x00...1',
+   *         registrationTypes = [RegistrationTypes.OnChain],
+   *     }],
+   * });
+   * ```
+   * or
+   * ```typescript
+   * claimsService.revokeMultipleClaim({
+   *     claimIds: ['245a40a9...776071ca57cec', '245a40a9...776071ca57cec'],
+   * });
+   * ```
+   *
+   * @param {RevokeMultipleClaimOptions} options object containing options
+   */
+  async revokeMultipleClaim({
+    claimIds,
+    claims,
+  }: RevokeMultipleClaimOptions): Promise<void> {
+    if (!claimIds || !claims) {
+      throw new Error(ERROR_MESSAGES.REVOKE_CLAIM_MISSING_PARAMETERS);
+    }
+
+    const revoker = this._signerService.did;
+    let claimsDetailsToRevoke: RevokeMultipleClaimOptions['claims'] = [
+      ...(claims || []),
+    ];
+    let namespace = claims[0]?.namespace;
+
+    if (claimIds) {
+      const claimsDetails = await Promise.all([
+        ...claimIds.map(async (claimId) => {
+          return await this.getClaimById(claimId);
+        }),
+      ]);
+      claimsDetailsToRevoke = [
```

Comment:
Is this means that claims should not be provided along with claimId's? If so we need to mention this in description or better exclude such possibility at all

Reply:
Yes, for all revocation method you can specify claim details or just simply pass claim id. And claim id will always override claim details. I've added comments to documentation and example uses